_posts/2019-07-11-deploy-spring-boot-kubernetes.md

```diff
@@ -0,0 +1,135 @@
+
+Assuming you have already heard of Kubernetes, you are probably aware of the continuing growth of this platform. More and more platforms are growing in popularity because of the flexibility of these technologies. Look at OpenShift, Cloud Foundry, ....
+
+As adaptation is growing, many developers are wondering how they effectively use this platform to deploy their application in the cloud with a Kubernetes cluster and use its benefits.
+
+Many big providers have already picked up Kubernetes and are providing their own implementations. A couple of examples are [Amazon Web Services (EKS)](https://aws.amazon.com/eks/){:target="_blank" rel="noopener noreferrer"}, [Google Cloud Platform (GKE)](https://cloud.google.com/kubernetes-engine/){:target="_blank" rel="noopener noreferrer"}, [Azure (AKS](https://azure.microsoft.com/nl-nl/services/kubernetes-service/){:target="_blank" rel="noopener noreferrer"}, [DigitalOcean](https://www.digitalocean.com/products/kubernetes/){:target="_blank" rel="noopener noreferrer"}, .... 
+
+## Prerequisites
+* [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/){:target="_blank" rel="noopener noreferrer"}
+* A Spring Boot project
+* A Kubernetes cluster in the cloud (you can use [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/){:target="_blank" rel="noopener noreferrer"} if you want to try this on your local machine)
+
+## First things first: creating a Docker image
+
+Kubernetes works with Docker images. This means that your application needs to be dockerized so it can be pushed to a Docker registry. You can find a sample Dockerfile below.
+
+```
+FROM openjdk:8-jre-alpine
+WORKDIR /tmp
+COPY target/app-0.0.1-SNAPSHOT.jar app.jar
+ENTRYPOINT ["java", "-jar", "app.jar"]
+```
+This is a very basic Dockerfile, but this should be enough to do what is needed.
+
+<img class="image right" alt="Docker" src="/img/2019-07-11-deploy-spring-boot-kubernetes/docker.png">
+
+The first line tells us to use the 8-jre-alpine image from the openJDK repository as base image.
+The second line tells the image that it should work from the /tmp directory.
+The third line copies the compiled JAR file from your target folder to the Docker image (you might have to rename the file name depending on the name of your project).
+
+
+Finally, we tell our image to use the java command as entry point, meaning that once the Docker image starts running, it has to execute that command.
+
+You can now push this image to your favorite Docker registry, as Kubernetes will need to pull this image from the registry later. If you do not have a Docker registry, I suggest using Docker Hub.
+
+## Where the magic happens: deploying your application on the Kubernetes cluster
+
+All Kubernetes configurations are written in YAML. This is because Kubernetes configuration files are meant to be easily readable by the human eye, and the Kubernetes team decided to use YAML instead of JSON.
```

Comment:
```suggestion
All Kubernetes configurations are written in YAML. The reason for this is that Kubernetes configuration files are meant to be easily readable by the human eye and the Kubernetes team decided to use YAML instead of JSON.
```